source/read/retrieve.txt

```diff
@@ -0,0 +1,154 @@
+In this guide, you can learn how to use the {+driver-short+} to retrieve data from a
+MongoDB collection by using read operations. You can call the ``find()`` method to
+retrieve documents that match a set of criteria specified in a query filter.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+.. _kotlin-sync-retrieve-find:
+
+Find Documents
+--------------
+
+The ``find()`` method retrieves documents from a collection. This
+method takes a **query filter** and returns all matching documents. A query filter is a
+document that specifies the criteria that the driver uses to match documents from the
+collection.
+
+.. To learn more about query filters, see :ref:`kotlin-sync-specify-query`.
+
+Find Documents Example
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following example uses the ``find()`` method to find all documents in which the
+value of the ``cuisine`` field is ``"Spanish"``:
+
+.. literalinclude:: /includes/read/retrieve.kt
+   :start-after: start-find
+   :end-before: end-find
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+The ``find()`` operation in the preceding example returns a ``FindIterable`` object,
+which you can iterate through by using the ``forEach()`` method, as shown in the following
+example:
+
+.. literalinclude:: /includes/read/retrieve.kt
+   :start-after: start-find-iterate
+   :end-before: end-find-iterate
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. note:: Find All Documents
+ 
+   To find all documents in a collection, pass an empty filter to the ``find()`` method:
+
+   .. literalinclude:: /includes/read/retrieve.kt
+      :start-after: start-find-all
+      :end-before: end-find-all
+      :language: kotlin
+      :copyable:
+      :dedent:
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` method by chaining methods to
+the ``find()`` method call. The following table describes commonly used methods used for
+modifying queries:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``batchSize()`` 
+     - | Limits the number of documents to return per batch. To learn more about
+         batch size, see :manual:`cursor.batchSize() </reference/method/cursor.batchSize/>`
+         in the MongoDB Server manual.
+
+   * - ``collation()`` 
+     - | An instance of the ``Collation`` class that sets the collation options. 
+
+   * - ``comment()`` 
+     - | A string to attach to the query. This can help you trace and interpret the
+         operation in the server logs and in profile data. To learn more about query comments, 
+         see :manual:`$comment </reference/operator/query/comment/>` in the MongoDB Server
+         manual.
```

Comment:
```suggestion
   * - ``comment()`` 
     - | Specifies a string to attach to the query. This can help you trace and interpret the
         operation in the server logs and in profile data. To learn more about query comments, 
         see :manual:`$comment </reference/operator/query/comment/>` in the MongoDB Server
         manual.
```

Reply:
and so on